### PR TITLE
feat!: Remove --save_resume and add --predict_only_model. Resolves: #3163

### PR DESCRIPTION
**Pull request by bassmang**
feat: Remove save_resume and add save_bare. Resolves: VowpalWabbit#3163

### CODE REVIEW

**Comment by jackgerrits** [comment]
What is the expected behavior when a user loads a 8.11.0 model which has save resume? And if it doesnt?

**Comment by bassmang** [comment]
@jackgerrits both of those models should be loaded in and have the same results as before. A new model with no flag would be the same as an old model with --save_resume, and a new model with --save_bare would be the same as an old model with no flag.

**Comment by jackgerrits** [comment]
Excellent, that sounds reasonable

**Comment by olgavrou** [comment]
> @jackgerrits both of those models should be loaded in and have the same results as before. A new model with no flag would be the same as an old model with --save_resume, and a new model with --save_bare would be the same as an old model with no flag.

what about a new model with no `--save_bar` (i.e. it has `save_resume` on) would it load fine and continue learning on an old vw instance?

**Comment by bassmang** [comment]
> > @jackgerrits both of those models should be loaded in and have the same results as before. A new model with no flag would be the same as an old model with --save_resume, and a new model with --save_bare would be the same as an old model with no flag.
> 
> what about a new model with no `--save_bar` (i.e. it has `save_resume` on) would it load fine and continue learning on an old vw instance?

Yes this works fine too and can continue learning -- just tested locally

**Review by olgavrou — APPROVED** [review]

File: vowpalwabbit/parse_args.cc [hunk 1]
```diff
@@ -1143,7 +1148,12 @@ void parse_output_model(options_i& options, VW::workspace& all)
   if (!all.final_regressor_name.empty() && !all.logger.quiet)
     *(all.trace_message) << "final_regressor = " << all.final_regressor_name << endl;
 
-  if (options.was_supplied("invert_hash")) all.hash_inv = true;
+  if (options.was_supplied("invert_hash")) { all.hash_inv = true; }
+  if (save_resume)
+  {
+    logger::errlog_warn("--save_resume flag is deprecated -- learning can now continue on saved models by default!");
```
Comment (jackgerrits) [inline vowpalwabbit/parse_args.cc:1154]: Small thing, I don't think the ! is necessary.
```suggestion
    logger::errlog_warn("--save_resume flag is deprecated -- learning can now continue on saved models by default.");
```

**Review by jackgerrits — APPROVED** [review]

**Pull request merged**